indico_toolkit/metrics/compare_ground_truth.py

```diff
@@ -0,0 +1,119 @@
+from indico_toolkit.association.association import sequences_overlap
+from indico_toolkit.types.extractions import Extractions
+from indico_toolkit.errors import ToolkitInputError
+
+
+class CompareGroundTruth:
+    """
+    Compare a set of ground truths against a set of model predictions on a per document basis.
+    """
+
+    def __init__(self, ground_truth: list, predictions: list):
+        self.gt_by_label: dict = Extractions(ground_truth).to_dict_by_label
+        self.preds_by_label: dict = Extractions(predictions).to_dict_by_label
+        self.labels: list[str] = list(
+            set(list(self.gt_by_label) + list(self.preds_by_label))
+        )
+        self.all_label_metrics: dict = None
+        self.overall_metrics: dict = None
+
+    def set_all_label_metrics(self, span_type: str = "overlap") -> None:
+        if span_type in ["exact", "overlap"]:
+            self.all_label_metrics = {
+                label: self._get_base_metrics(label, span_type) for label in self.labels
+            }
+        else:
+            raise ToolkitInputError("Invalid span type entered.")
+
+    def set_overall_metrics(self) -> None:
+        if self.all_label_metrics is None:
+            self.set_all_label_metrics()
+
+        metrics_types = [
+            "true_positives",
+            "false_positives",
+            "false_negatives",
+        ]
+
+        self.overall_metrics = {
+            "true_positives": 0,
+            "false_positives": 0,
+            "false_negatives": 0,
+        }
+
+        for metric in metrics_types:
+            total_amt = 0
+            for label in self.all_label_metrics:
+                total_amt += self.all_label_metrics[label][metric]
+                self.overall_metrics[metric] = total_amt
```

Comment:
This just gets overwritten after every iteration, you should set it equal to total_amt after the inner loop